retworkx/__init__.py

```diff
@@ -593,3 +593,39 @@ def _digraph_core_number(graph):
 @core_number.register(PyGraph)
 def _graph_core_number(graph):
     return graph_core_number(graph)
+
+
+def networkx_converter(graph):
+    """Convert a networkx graph object into a retworkx graph object.
+
+    .. note::
+
+        networkx is **not** a dependency of retworkx and this function
+        is provided as a convenience method for users of both networkx and
+        retworkx. This function will not work unless you install networkx
+        independently.
+
+    :param networkx.Graph graph: The networkx graph to convert.
+
+    :returns: A retworkx graph, either a :class:`~retworkx.PyDiGraph` or a
+        :class:`~retworkx.PyGraph` based on whether the input graph is directed
+        or not.
+    :rtype: :class:`~retworkx.PyDiGraph` or :class:`~retworkx.PyGraph`
+    """
+    if graph.is_directed():
+        if graph.is_multigraph():
+            new_graph = PyDiGraph()
+        else:
+            new_graph = PyDiGraph(multigraph=False)
+    else:
+        if graph.is_multigraph():
+            new_graph = PyGraph()
+        else:
+            new_graph = PyGraph(multigraph=False)
```

Comment:
This can be simplified
```suggestion
    if graph.is_directed():
        new_graph = PyDiGraph(multigraph=graph.is_multigraph())
    else:
        new_graph = PyGraph(multigraph=graph.is_multigraph())
```